### PR TITLE
Add tests for loading empty documents

### PR DESCRIPTION
**Pull request by tomalec**
Few more passing tests.
Asserts  webcomponents/webcomponentsjs/issues/535

Once again upgraded PR for V0
  - webcomponents/webcomponentsjs/pull/633
  - webcomponents/webcomponentsjs/pull/536

### CODE REVIEW

File: tests/html/load-empty.html [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    <script src="../../../web-component-tester/browser.js"></script>
+    <script>
+      window.loadEvents = 0;
+      (function() {
+        function importLoaded(event) {
+          window.loadEvents++;
+          chai.assert.equal(event.type, 'load', 'event type should be `load`');
+          chai.assert.ok(event.target.import, 'link should have `import` defined');
+          // https://github.com/webcomponents/html-imports/issues/3
+          // chai.assert.ok(event.target.import instanceof Document, 'load event target can be used to find imported document');
```
Comment (valdrinkoshi) [inline tests/html/load-empty.html]: (no action needed here) In the new polyfill we're experimenting the benefits of not having a separate Document created for each import, but rather just an Element. We can further discuss in the issue #3

File: tests/html/load-empty.html [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      test('load empty resource asynchronously', function(done) {
+        loadEvents = 0;
+        setTimeout(function(){
+          var anotherHTMLImport = document.createElement('link');
+          anotherHTMLImport.rel = 'import';
+          anotherHTMLImport.href = 'imports/load-empty-async.html';
+          anotherHTMLImport.onload = importLoaded;
+          document.head.appendChild(anotherHTMLImport);
+
+          setTimeout(function(){
```
Comment (valdrinkoshi) [inline tests/html/load-empty.html]: Here and below you could rely on the details passed in the callback of `whenReady` instead of using `setTimeout`
```js
  HTMLImports.whenReady(function(detail) {
    assert.equal(detail.allImports.length, 2);
    assert.equal(detail.loadedImports.length, 2);
    assert.equal(detail.errorImports.length, 0);
    done();
  });

File: tests/html/load-empty.html [hunk 3]
```diff
@@ -0,0 +1,77 @@
+<!doctype html>
+<!--
+    @license
+    Copyright (c) 2014 The Polymer Project Authors. All rights reserved.
```
Comment (valdrinkoshi) [inline tests/html/load-empty.html]: Please replace with `(c) 2017` 👌

**Review by valdrinkoshi — CHANGES_REQUESTED** [review]
Thanks for the contribution @tomalec! Minor comments

File: tests/html/load-empty.html (hunk 2 above)
Comment (tomalec) [reply tests/html/load-empty.html]: Thanks for the review :)

Naturally, I can change that. But after second thought, I started to doubt: 
Wouldn't it make the individual tests dependent on each other? We would test two separate features at once, making them not to "unit".

Comment (valdrinkoshi) [reply tests/html/load-empty.html]: Yes, it would make the tests dependent between them because you'd get the cumulated total # of imports.
I was a bit skeptical for the `setTimeout` of 100ms which is a bit random value, we could just add a listener on load and call `done()` in the callback, but that's a nit.
Let's keep these tests units 👌 This LGTM

**Review by valdrinkoshi — APPROVED** [review]

**Comment by tomalec** [comment]
Maybe adding listener for `load` event that we are checking is not that bad idea, in case of a failure, the test will timeout (after 1000ms which is not so random - it's a global timeout for every test to execute)

**Comment by tomalec** [comment]
I have prepared a change with `load` event here https://github.com/tomalec/html-imports/commit/83a643829a88842d9858454609e86a286305c8f1 but it does not check whether `onload` was called more times that desired, so I'm not sure which one is better.

**Comment by valdrinkoshi** [comment]
You could use `elem.addEventListener('load', function(event)` and keep those separated.
Also, wct will report when `done` is called multiple times for a single test 👌

**Comment by valdrinkoshi** [comment]
This LGTM, thanks again for the contribution @tomalec! 💯

**Comment by tomalec** [comment]
> Also, wct will report when `done` is called multiple times for a single test

Oh, that's cool, I was not aware of it.


Thanks for support @valdrinkoshi :)
